src/msix/pack/AppxPackageWriter.cpp

```diff
@@ -149,22 +148,35 @@ namespace MSIX {
         // TODO: use memoryLimit for how many files are going to be added
         for(UINT32 i = 0; i < fileCount; i++)
         {
-            ThrowErrorIf(Error::InvalidParameter, IsFootPrintFile(payloadFiles[i].fileName), "Trying to add footprint file to package");
             ComPtr<IStream> stream(payloadFiles[i].inputStream);
-            ProcessFileAndAddToPackage(payloadFiles[i].fileName, stream, payloadFiles[i].compressionOption, payloadFiles[i].contentType, false);
+            ProcessFileAndAddToPackage(payloadFiles[i].fileName, stream, payloadFiles[i].compressionOption, payloadFiles[i].contentType, FileType::AppxPayload);
         }
         failState.release();
         return static_cast<HRESULT>(Error::OK);
     } CATCH_RETURN();
 
     void AppxPackageWriter::ProcessFileAndAddToPackage(const std::string& name, const ComPtr<IStream>& stream,
-        APPX_COMPRESSION_OPTION compressionOpt, const char* contentType, bool forceContentTypeOverride, bool addToBlockMap)
+        APPX_COMPRESSION_OPTION compressionOpt, const char* contentType, FileType fileType)

```

Comment:
>FileType fileType [](start = 73, length = 17)

This feels more cumbersome than before, although I get that it added some new information you needed.

Could we instead use an overload strategy?